docs: connect-service-upstreams annotation fixes (#17312)
hashicorp/consul: website/content/docs/k8s/annotations-and-labels.mdx

- `consul.hashicorp.com/connect-service-upstreams` - The list of upstream
  services that this pod needs to connect to via the service mesh along with a static
  local port to listen for those connections. When transparent proxy is enabled,
  this annotation is optional. This annotation can be either _unlabeled_ or _labeled_. Unlabeled annotations use component names, and the component types are inferred through the annotation's syntax. Labeled annotations explicitly associate component names and types to form Consul DNS addresses. To configure upstreams for [cluster peering](/consul/docs/k8s/connect/cluster-peering/tech-specs), use the labeled format.

    - **Unlabeled**:

>        Use the unlabeled annotation format to specify a service name, Consul Enterprise namespaces and partitions, and

Reviewer: This pairs with the suggested changes to the labeled section.

```suggestion
       The unlabeled annotation format allows you to reference any service not in a cluster peer as an upstream. You can specify a Consul Enterprise namespace. You can also specify an admin partition in the same datacenter or a WAN federated datacenter. Unlike the labeled annotation, you can also reference a prepared query as an upstream.
```

I hope to publish some changes to the docs that include an explanation on why prepared queries are not recommended with service mesh... but because that content isn't available to link to yet, I don't want to try to explain why prepared queries aren't recommended as upstreams here. I'll address that in my (future) PR.

       datacenters. 

>       - Service name: Place the service name at the beginning of the annotation to specify the upstream service. You also have the option to append the datacenter where the service is deployed.

Reviewer: ```suggestion
      - Service name: Place the service name at the beginning of the annotation to specify the upstream service. You also have the option to append the WAN federated datacenter where the service is deployed.
```

   
        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name]:[port]:[optional datacenter]"
        ```

      - Namespace (requires Consul Enterprise 1.7+): Upstream services may be running in a different namespace. Place
      the upstream namespace after the service name. For additional details about configuring the injector, refer to [Consul Enterprise namespaces](#consul-enterprise-namespaces) .
        
        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].[service-namespace]:[port]:[optional datacenter]"
        ```

        If the namespace is not specified, the annotation defaults to the namespace of the source service.

>         If your version of Consul Enterprise in lower than v1.7, Consul interprets the value placed in the namespace position as part of the service name.

Reviewer 2: ```suggestion
        Consul Enterprise v1.7 and older interprets the value placed in the namespace position as part of the service name.
```
Did we decide on referring to versions in temporal or cardinal order?

Reviewer: Tangential: Do we still need explanations like this now that we have versioned docs? Or should we remove things that talk about differences in behavior in fairly old versions?

It's not something we have to decide for this PR. It probably makes sense to make a decision and then apply across the docs surface if needed rather than piecemeal in each PR.

      - Admin partitions (requires Consul Enterprise 1.11+): Upstream services may be running in a different
        partition. When specifying a partition, you must also specify a namespace. Place the partition name after the namespace. If you specify the name of the datacenter, it must be the local datacenter. Communicating across partitions using this method is only supported within a
        datacenter. For cross partition communication across datacenters, [establish a cluster
        peering connection](/consul/docs/k8s/connect/cluster-peering/usage/establish-peering) and set the upstream with a labeled annotation format.

        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].[service-namespace].[service-partition]:[port]:[optional datacenter]"
        ```
        
      - Prepared queries: To configure a [prepared query](/consul/api-docs/query) in an upstream annotation, prepend the annotation
        with `prepared_query` and then invoke the name of the query.

        ```yaml
        annotations:
          'consul.hashicorp.com/connect-service-upstreams': 'prepared_query:[query name]:[port]'
        ```

>     - **Labeled** (requires Consul for Kubernetes v0.45.0+):

Reviewer: Move this above the **Unlabeled** section

>     - **Labeled** (requires Consul for Kubernetes v0.45.0+):

Reviewer: ```suggestion
    - **Labeled** (requires Consul on Kubernetes v0.45.0+):
```

>       The labeled annotation format is required when specifying an upstream service in a different datacenter. You can specify a peer, namespace, partition, or datacenter, but the format supports only one peer, datacenter, or partition per annotation.

Reviewer: If we change the text above to generally recommend the labeled format, then it seems unnecessary to say that this is required when specifying an upstream in a different datacenter. It also suggests that unlabeled formats can't reference upstreams in a different datacenter---they can, as long as it is a WAN federated datacenter (not a cluster peer).

I'm not sure what the original text of "supports only one peer, datacenter, or partition" means. I think it's intended to mean that you can only specify one of those fields per upstream. I worry that the wording of "per annotation" can give a false impression that the `connect-service-upstreams` annotation can only have one upstream.

Perhaps something like:
```suggestion
      The labeled annotation format allows you to reference any service as an upstream. You can specify a Consul Enterprise namespace. You can also specify an admin partition in the same datacenter, a cluster peer, or a WAN federated datacenter.
```

      - Service name: Place the service name at the beginning of the annotation followed by `.svc` to specify the upstream service.

        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc:[port]"
        ```
      
      - Peer or datacenter: Place the peer or datacenter after `svc.` followed by either `peer` or `dc` and the port number.
        
        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-peer].peer:[port]"
        ```
        
        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-dc].dc:[port]"
        ```
      
      - Namespace (requires Consul Enterprise): Place the namespace after `svc.` followed by `ns` and the port number.
        
        ```yaml
        annotations:
          "consul.hashicorp.com/connect-service-upstreams":"[service-name].svc.[service-namespace].ns:[port]"
        ```
        

>         When namespaces are enabled, you must include the namespace in the annotation before specifying a peer, datacenter, or admin partition.

Reviewer: ```suggestion
        When namespaces are enabled, you must include the namespace in the annotation before specifying a cluster peer, WAN federated datacenter, or admin partition in the same datacenter.
```
